README.md

```diff
@@ -16,5 +17,60 @@ For now, repository contains:
   2. can serialize your AST to and from s-expressions, either to look at it, or for interchange.
 
 Outside of those notable classes and some utilities, there are 2 important subfolders:
+
 - `sexpr`: facilities for reading and writing S-Expressions. Mostly for use by Wellformed, but tested on a necessary-seeming subset of the original proposed RFC.
 - `tla`: under-construction parser for the TLA+ modeling language, which has been invaluable in stress-testing all the generic machinery. The test suite involves trying to parse the full TLA+ Examples folder.
+
+# Getting Started
+
+Follow these steps to setup and test the project:
+
+1. **Clone the Repository**  
+   Clone this repository to your local machine:
+
+```bash
+git clone <https://github.com/DistCompiler/dcal.git>
+```
+
+2. **Install Scala CLI**
+   Windows: Install Scala CLI using Winget:
+
+```bash
+winget install virtuslab.scalacli
+```
+
+Linux: Install Scala CLI using curl:
+
+```bash
+curl -sSLf https://scala-cli.virtuslab.org/get | sh
+```
+
+MacOS: Install Scala CLI using Homebrew:
+
+```bash
+brew install Virtuslab/scala-cli/scala-cli
+```
+
+3. **Verify Installation**
+   After installation, close and reopen your terminal, then verify Scala CLI is installed by running:
+
+```bash
+scala-cli version
+```
+
+4. **Building the System**
+   After veryifying installation, build the compiler by running:
+
+```bash
+scala-cli
```

Comment:
Not sure if this does what you think it does. You can actually skip it and go straight to test; it auto-compiles first.

Refreshing my memory with this docs page https://scala-cli.virtuslab.org/docs/commands/basics, I think the behavior in our case is to open a REPL, which might not compile the code at all.

Anyway, the correct compile command should be `scala-cli compile .` imo

As we discussed before, mentioning the Metals VSCode plugin is key for anyone who hasn't seen a Scala dev setup before. That is also a good way to compile the code.